model/version.go

```diff
@@ -155,6 +156,36 @@ func (self *Version) SetNotActivated() error {
 	)
 }
 
+func (self *Version) SetAborted() error {
+	self.Aborted = utility.TruePtr()
+	return VersionUpdateOne(
+		bson.M{VersionIdKey: self.Id},
+		bson.M{
+			"$set": bson.M{
+				VersionAbortedKey: true,
+			},
+		},
+	)
+
+}
+
+func (self *Version) SetNotAborted() error {
+	self.Aborted = utility.FalsePtr()
+	return VersionUpdateOne(
+		bson.M{VersionIdKey: self.Id},
+		bson.M{
+			"$set": bson.M{
+				VersionAbortedKey: false,
+			},
+		},
+	)
+
+}
+
+func (self *Version) IsAborted() bool {
+	return utility.FromBoolPtr(self.Aborted)
```

Comment:
This function doesn't quite work; if t.Aborted == nil, then would we need to query all tasks and then set it?